src/utils/__tests__/webcrypto.test.js

```diff
@@ -64,13 +66,33 @@ describe('util/webcrypto', () => {
                     },
                 };
 
-                const expectedError = new Error('lol');
+                const expectedError = new Error('ERROR');
 
                 digest(algorithm, buffer).catch(error => {
                     expect(error).toBe(expectedError);
                 });
 
                 cryptoOperation.onerror(expectedError);
+                expect.assertions(1);
+            });
+        });
+        describe('js-sha1', () => {
+            test('should use js-sha1 for calculating hash in IE-11 SHA-1 digest scenarios', async () => {
+                // ie11 does not support sha-1, so we use a library
+                sha1.arrayBuffer = jest.fn().mockImplementation(() => new ArrayBuffer());
+                const digestMock = jest.fn().mockReturnValueOnce({});
+                window.crypto = undefined;
+
+                window.msCrypto = {
+                    subtle: {
+                        digest: digestMock,
+                    },
+                };
+
+                const hash = await digest('SHA-1', buffer);
+                expect(hash).toBeDefined();
+                expect(digestMock).not.toHaveBeenCalled();
+                expect(sha1.arrayBuffer).toHaveBeenCalledWith(buffer);
```

Comment:
Can we also test (1) that `sha1.arrayBuffer()` is not called when the algorithm is not SHA-1 and (2) that the promise is rejected when the call to `sha1.arrayBuffer1()` fails?

Reply:
added tests